adi/ad7124.py

```diff
@@ -0,0 +1,116 @@
+#     - Redistributions in binary form must reproduce the above copyright
+#       notice, this list of conditions and the following disclaimer in
+#       the documentation and/or other materials provided with the
+#       distribution.
+#     - Neither the name of Analog Devices, Inc. nor the names of its
+#       contributors may be used to endorse or promote products derived
+#       from this software without specific prior written permission.
+#     - The use of this software may or may not infringe the patent rights
+#       of one or more patent holders.  This license does not release you
+#       from the requirement that you obtain separate licenses from these
+#       patent holders to use this software.
+#     - Use of the software either in source or binary form, must be run
+#       on or directly connected to an Analog Devices Inc. component.
+#
+# THIS SOFTWARE IS PROVIDED BY ANALOG DEVICES "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+# INCLUDING, BUT NOT LIMITED TO, NON-INFRINGEMENT, MERCHANTABILITY AND FITNESS FOR A
+# PARTICULAR PURPOSE ARE DISCLAIMED.
+#
+# IN NO EVENT SHALL ANALOG DEVICES BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+# EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, INTELLECTUAL PROPERTY
+# RIGHTS, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR
+# BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+# STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+# THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+from decimal import Decimal
+
+import numpy as np
+from adi.attribute import attribute
+from adi.context_manager import context_manager
+from adi.rx_tx import rx
+
+
+class ad7124(rx, context_manager):
+    """ AD7124 ADC """
+
+    _complex_data = False
+    channel = []
+    _device_name = ""
+
+    def __init__(self, uri=""):
+
+        context_manager.__init__(self, uri, self._device_name)
+        self._ctrl = self._ctx.find_device("ad7124-8")
+        self._rxadc = self._ctx.find_device("ad7124-8")
+
+        # dynamically get channels
+        for ch in self._ctrl._channels:
+            name = ch._id
+            self._rx_channel_names.append(name)
+            self.channel.append(self._channel(self._ctrl, name))
+        rx.__init__(self)
+
+    @property
+    def sampling_frequency(self):
+        # only works with the same frequency for all channels
+        return self._get_iio_attr(self.channel[0].name, "sampling_frequency", False)
+
+    @sampling_frequency.setter
+    def sampling_frequency(self, value):
+        for ch in self.channel:
+            self._set_iio_attr(ch.name, "sampling_frequency", False, value)
+        return 1
+
+    @property
+    def scale_available(self):
+        return self._get_iio_attr(self.channel[0].name, "scale_available", False)
+
+    class _channel(attribute):
```

Comment:
This is pretty handy to have. Do you think we could possibly combine this into the rx_tx class so data received from buffers are automatically converted to specific units? Say we had a property like rx_output_type = 'SI' or something?

The adis16460 class would benefit.

Reply:
Do you think this should be a separate pull request ?

Reply:
I added another commit that handles _rx_output_type .. seems to work for AD7124. Does the implementation make sense. I don't have a ADIS16460. I can probably test it tomorrow if I find one in the office.

Reply:
There is one inside the PackRF